### PR TITLE
[Vue] remove angularjs from Installation plugin

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

**This PR is based off of #19418.**

Changes:
* Convert twig contents for system check page to Vue components.
* Allow ContentIntro directive to be applied conditionally.
* Make Diagnostic result entities JSON serializable.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/Diagnostics/Diagnostic/RequiredPhpSetting.php [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 namespace Piwik\Plugins\Diagnostics\Diagnostic;
 
-class RequiredPhpSetting
+use JetBrains\PhpStorm\Internal\TentativeType;
+
```
Comment (sgiehl) [inline plugins/Diagnostics/Diagnostic/RequiredPhpSetting.php]:
```suggestion
```

File: plugins/Installation/Controller.php [hunk 2]
```diff
@@ -567,15 +570,17 @@ public function getInstallationJs()
...
+        $files[] = AssetManager\UIAssetFetcher\PluginUmdAssetFetcher::getUmdFileToUseForPlugin('CoreHome');
+        $files[] = AssetManager\UIAssetFetcher\PluginUmdAssetFetcher::getUmdFileToUseForPlugin('Installation');
 
         if (defined('PIWIK_TEST_MODE') && PIWIK_TEST_MODE
             && file_exists(PIWIK_DOCUMENT_ROOT . '/tests/resources/screenshot-override/override.js')) {
             $files[] = 'tests/resources/screenshot-override/override.js';
         }
 
-        return AssetManager::compileCustomJs($files);
+        $result = StaticContainer::get('Piwik\Translation\Translator')->getJavascriptTranslations() . "\n";
```
Comment (sgiehl) [inline plugins/Installation/Controller.php]: We actually can't include the javascript translations here. The problem is, that it is possible to change the language in the drop down. Doing that will change the language, but the string contained in javascript will still be in the initially loaded language, as the javascript is cached.

**Review by sgiehl — CHANGES_REQUESTED** [review]
Left some comments. Clicked through installation process and everything else still seems to work as expected

**Comment by diosmosis** [comment]
@sgiehl applied fixes

**Review by sgiehl — APPROVED** [review]

**Pull request merged**